SimpleWAWS/Code/BackgroundQueueManager.cs

```diff
@@ -191,7 +233,20 @@ private void MaintainResourceGroupLists()
 
             foreach (var resource in resources)
                 DeleteResourceGroup(resource);
-            //TODO:Add a way replenish leaking resourcegroups
+
+            // delete duplicate/leaking resource groups
+            var duplicateResourceGroups = this.FreeResourceGroups
+            .Where(rg => (rg.IsSimpleWAWSResourceName))
+            .GroupBy(s => s.GeoRegion)
+            .SelectMany(g => g.Skip(1))
+            .Union(
+             this.FreeJenkinsResourceGroups
+            .Where(rg => (rg.IsSimpleWAWSResourceName))
+            .GroupBy(s => s.GeoRegion)
+            .SelectMany(g => g.Skip(SimpleSettings.JenkinsResourceGroupsPerRegion)));
+
+            foreach (var resource in duplicateResourceGroups)
+                DeleteDuplicateResourceGroup(resource);
```

Comment:
You are right. But I do see the same resourcegroup in the `FreeResourceGroups` twice eg. `/subscriptions/11415693-3da2-4784-9e4c-b36c7d7e28f7/resourceGroups/TRY_RG_Canada_East_7dbe7cfe-b4c2-46d1-a8a8-f8721ec33fb0 ` . I'll remove this portion and strengthen the check when it gets enqueued.